types/node/test/test.ts

```diff
@@ -911,3 +912,22 @@ const invalidTestContext = new TestContext();
 
 // @ts-expect-error Should not be able to instantiate a SuiteContext
 const invalidSuiteContext = new SuiteContext();
+
+test('planning with streams', (t, done) => {
+    function* generate() {
+      yield 'a';
+      yield 'b';
+      yield 'c';
+    }
+    const expected = ['a', 'b', 'c'];
+    t.plan(expected.length);
+
+    const stream = Readable.from(generate());
+    stream.on('data', (chunk) => {
+        t.assert.strictEqual(chunk, expected.shift());
```

Comment:
> Assertions require every name in the call target to be declared with an explicit type annotation  @definitelytyped/expect

https://github.com/microsoft/TypeScript/issues/36931

Not sure how to fix this...

Reply:
Hi @G-Rath! I see that you added assertion guards in assert module (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42786), maybe you can help with this.

Reply:
This seems unavoidable for any object method that returns an assertion, regardless of exactly how it's defined. To use these methods, any callback provided to `test()` would need to explicitly type the callback parameter to make the compiler happy.

```ts
test(t => { t.assert.strictEqual(true, true) }); // error
test((t: TestContext) => { t.assert.strictEqual(true, true) }); // ok
```

The only alternative would be to duplicate the definition from `node:assert` but type the return as `void` instead of `asserts ...`, but that feels even less satisfactory.

Could perhaps just add this warning to the docblock.

Reply:
This seems unavoidable for any object method that returns an assertion, regardless of exactly how it's defined. To use these methods, any callback provided to `test()` would need to explicitly type the callback parameter to make the compiler happy.

```ts
test(t => { t.assert.strictEqual(true, true) }); // error
test((t: TestContext) => { t.assert.strictEqual(true, true) }); // ok
```

The only alternative would be to duplicate the definition from `node:assert` but type the return as `void` instead of `asserts ...`, but that feels even less satisfactory.

Could perhaps just add this warning to the docblock.